src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -0,0 +1,1030 @@
+        public NativeWindow(INativeWindowProperties settings)
+        {
+            _title = settings.Title;
+
+            unsafe
+            {
+                switch (settings.WindowBorder)
+                {
+                    case WindowBorder.Hidden:
+                        Glfw.WindowHint(WindowHintBool.Decorated, false);
+                        break;
+
+                    case WindowBorder.Resizable:
+                        Glfw.WindowHint(WindowHintBool.Resizable, true);
+                        break;
+
+                    case WindowBorder.Fixed:
+                        Glfw.WindowHint(WindowHintBool.Resizable, false);
+                        break;
+                }
+
+                _windowBorder = settings.WindowBorder;
+
+                if (settings.WindowState == WindowState.Fullscreen)
+                {
+                    var monitor = settings.CurrentMonitor.ToUnsafePtr<GraphicsLibraryFramework.Monitor>();
+                    var modePtr = Glfw.GetVideoMode(monitor);
+                    Glfw.WindowHint(WindowHintInt.RedBits, modePtr->RedBits);
+                    Glfw.WindowHint(WindowHintInt.GreenBits, modePtr->GreenBits);
+                    Glfw.WindowHint(WindowHintInt.BlueBits, modePtr->BlueBits);
+                    Glfw.WindowHint(WindowHintInt.RefreshRate, modePtr->RefreshRate);
+                    WindowPtr = Glfw.CreateWindow(modePtr->Width, modePtr->Height, _title, monitor, null);
+                }
+                else
+                {
+                    WindowPtr = Glfw.CreateWindow(settings.Width, settings.Height, _title, null, null);
+                }
+
+                RegisterWindowCallbacks();
+
+                Focused = settings.Focused;
+                Visible = settings.Visible;
+                WindowState = settings.WindowState;
+
+                IsEventDriven = settings.IsEventDriven;
+
+                if (settings.X > -1)
+                {
+                    X = settings.X;
+                }
+
+                if (settings.Y > -1)
+                {
+                    Y = settings.Y;
+                }
+
+                Glfw.GetFramebufferSize(WindowPtr, out var width, out var height);
+                ClientSize = new Vector2(width, height);
+            }
+
+            Interlocked.Increment(ref NumberOfUsers);
+        }
+
+        private void RegisterWindowCallbacks()
+        {
+            unsafe
+            {
+                Glfw.SetWindowPosCallback(WindowPtr, (window, x, y) => OnMove(this, new WindowPositionEventArgs(x, y)));
+
+                Glfw.SetWindowSizeCallback(WindowPtr, (window, width, height) => OnResize(this, new ResizeEventArgs(width, height)));
+
+                Glfw.SetWindowCloseCallback(WindowPtr, window => OnClosing(this, new CancelEventArgs()));
+
+                Glfw.SetWindowIconifyCallback(WindowPtr, (window, iconified) => OnIconChanged(this, new IconifyEventArgs(iconified)));
```

Comment:
Found this big mistake here. Iconification is not when the window gets a new Icon, but when it is e.g. minimized or restored or similar things(dependent on WindowManager minimization is not available, but other alternatives instead)

Reply:
Crud, I thought I'd fixed every instance of this.

Reply:
Fixed this. Maybe we should change "Iconified" to "Minimized" to avoid confusion?

Reply:
I'd prefer if we kept the GLFW terms for parity with their documentation.

Reply:
But if you've already changed it to minimized that's fine - it doesn't matter all that much.

Reply:
It's still the same on the GLFW bindings, but this is the first time I've ever heard it be called "Iconification". I just think that using a less common term like that might cause some confusion.

Reply:
Can't disagree with that. Keep it up! things are looking great.

Reply:
Thanks! (Can you resolve this, by the way? I can't because I didn't open the PR)